src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -211,10 +211,12 @@ private ConstantValue DefaultSyntaxValue
                     var completedOnThisThread = state.NotePartComplete(CompletionPart.EndDefaultSyntaxValue);
                     Debug.Assert(completedOnThisThread);
 
-                    if (binder is not null && parameterEqualsValue is not null && !_lazyDefaultSyntaxValue.IsBad)
+                    if (binder is not null &&
+                        !_lazyDefaultSyntaxValue.IsBad &&
+                        parameterEqualsValue is { Syntax: EqualsValueClauseSyntax { Value: { } valueSyntax } })
```

Comment:
>parameterEqualsValue is { Syntax: EqualsValueClauseSyntax { Value: { } valueSyntax } } [](start = 24, length = 86)

Addition of this condition doesn't feel appropriate for the purpose of ```VerifyParamDefaultValueMatchesAttributeIfAny``` check. At least, we didn't care about specific node kind before and used to get syntax differently. It is not obvious that we are always going to end up with the same syntax node. If we can make an assumption that we are, perhaps we should simply assert the fact rather than changing the logic.